forcats.Rmd

```diff
@@ -0,0 +1,90 @@
+# Categorical data: forcats {#forcats}
+
+## Introduction
+This chapter is dedicated to the handling of categorical variables. This becomes important if information is to be presented in a non-alphabetical order or aggregated in a meaningful way. Within the R programming language, categorical variables are converted into a form that can be used for analysis using factors. While many Base-R functions automatically convert character vectors into factors, tidyverse requires an explicit treatment. The core-tidyverse provides the package forcats, which will be described here. Further information and exercises are available at the sources shown.
+
+Resources:
+
+- [Homepage](https://forcats.tidyverse.org/)
+- [Cheatsheet](https://github.com/rstudio/cheatsheets/raw/master/factors.pdf)
+- [Chapter in R4DS](https://r4ds.had.co.nz/factors.html)
+- [Vignette](https://forcats.tidyverse.org/articles/forcats.html)
+
+In this chapter the demo data set "diamonds" from the ggplot2-package ([more information](https://ggplot2.tidyverse.org/reference/diamonds.html)) and a dataset "olympic-games" which shows the medal success of olympic athletes from 1896 to 2016 are used. Before you start reading, you should have read the chapter ***Wrangling Data: dplyr*** or already have basic knowledge in this field. 
+
+```{r message=FALSE, warning=FALSE, paged.print=TRUE}
+library(tidyverse)
+diamonds<-read_csv("data/diamonds.csv")
+olympic<-read_csv("data/olympic-games.csv")
+
+head(diamonds)
+head(olympic)
+```
+
+## General functions
+### Create
+Basically two things are needed to create a factor. A vector which contains the values to be analyzed and a vector which divides the values into levels. As an example we will use the column "clarity" of the diamonds dataset. It is a categorical evaluation of the clarity of a diamond (I1 (worst), SI2, SI1, VS2, VS1, VVS2, VVS1, IF (best)). If you are interested in the distribution of the diamonds in this category, you could do this by using a suitable query:
+
+```{r}
+diamonds%>%
+  group_by(clarity)%>%
+  count()%>%
+  ggplot(aes(clarity, n))+
+  geom_col()
+```
+
+A sorting of the x-axis, which is based on the degree of clarity, is unfortunately not possible in this form. A workaround is to convert this column into a factor, which allows us to evaluate the individual categories. For this purpose we first define a vector, which ranks the categories according to their value:
+
+```{r}
+levels_clarity<-c("I1", "SI2", "SI1", "VS2", "VS1", "VVS2", "VVS1", "IF")
+```
+
+In the next step we lay this newly created levels vector on the column "clarity" of our data set to create a factor. This is a base-R function.
```

Comment:
Make clear which function is a base-R function.

Reply:
Right, that's misleading. I'll solve this